nbconvert/exporters/pdf.py

```diff
@@ -154,41 +157,78 @@ def log_error(command, out):
+            true_output_path = resources.get('metadata', {}).get('path', cwd)
+            resources['output_files_dir'] = os.path.join(td, 'outputs')
+            resources['metadata'] = resources.get('metadata', {})
+            resources['metadata']['path'] = resources['output_files_dir']
+            # We also need to trick the writer to match the temp output dir
+            self.writer.relpath = resources['output_files_dir']
+            self.writer.build_directory = resources['output_files_dir']
+            # Ensure the directory exists ahead of time
+            print(resources['output_files_dir'])
+            self.writer._makedir(resources['output_files_dir'])
+
+            try:
+                yield
+            finally:
+                # Copy any new files back
+                for f in os.listdir(resources['output_files_dir']):
+                    shutil.move(os.path.join(resources['output_files_dir'], f), true_output_dir)
+                resources['output_files_dir'] = true_output_dir
+                resources['metadata']['path'] = true_output_path
+                # Reset the writer
+                self.writer.relpath = true_output_path
+                self.writer.build_directory = true_output_path
+
     def from_notebook_node(self, nb, resources=None, **kw):
-        latex, resources = super(PDFExporter, self).from_notebook_node(
-            nb, resources=resources, **kw
-        )
-        # set texinputs directory, so that local files will be found
-        if resources and resources.get('metadata', {}).get('path'):
-            self.texinputs = resources['metadata']['path']
-        else:
-            self.texinputs = getcwd()
-        
-        self._captured_outputs = []
-        with TemporaryWorkingDirectory():
+        if not resources:
+            resources = {}
+        with self._fake_all_output_paths(resources):
+            latex, resources = super(PDFExporter, self).from_notebook_node(
+                nb, resources=resources, **kw
+            )
+            # set texinputs directory, so that local files will be found
+            if resources and resources.get('metadata', {}).get('path'):
+                self.texinputs = resources['metadata']['path']
+            else:
+                self.texinputs = getcwd()
+
+            self._captured_outputs = []
             notebook_name = 'notebook'
             tex_file = self.writer.write(latex, resources, notebook_name=notebook_name)
             self.log.info("Building PDF")
             rc = self.run_latex(tex_file)
             if rc:
-                rc = self.run_bib(tex_file)
-            if rc:
+                # This can fail for non-conversion blocking reasons
+                self.run_bib(tex_file)
                 rc = self.run_latex(tex_file)
-            
+            if not rc:
+                raise RuntimeError('Failed to run "{}" commands'.format(
+                    [cmd.format(filename=tex_file) for cmd in self.latex_command]))
```

Comment:
This won't display the captured output, will it? I think displaying the captured output is necessary if this is to fix #896 / #947. Otherwise, the user will still have zero idea why latex failed.

Reply:
It does emit a critical log that goes to stderr by default. I didn't put the full output in the error message because it can be 100's or 1000's or lines long when I was testing. I'll look at truncating the output and putting a piece of it here so there's some visibility if log messages aren't being streamed to the client for some reason.

Reply:
#988 Will fix latex error messages being far too long and not user friendly. I think that along with the fix to the notebook2.ipynb (which I see you did here), and the check for latex's return code beside the check for the file (like in #947) should nicely return any latex error and resolve #896.

Reply:
I'll rip out the message trimming I put in after #988 merges. I think this PR now implements #947 -- by raising on latex failure inside the `run_command` -- which the author hasn't been updating for the past week anyway.

Reply:
Ah, I didn't see the fact that you moved the latex failure into ```run_command```. I like that.